index.html

```diff
@@ -650,33 +650,52 @@ <h3 id="requesting-more-permission">
         </h3>
         <p>
           To <dfn data-lt="request permission to use|requesting permission to use" class=
-          "export">request permission to use</dfn> a |descriptor:PermissionDescriptor|, the user
-          agent must perform the following steps. This algorithm returns either
-          {{PermissionState/"granted"}} or {{PermissionState/"denied"}}.
+          "export">request permission to use</dfn> a |descriptor:PermissionDescriptor|, with
+          optionally a |promise:Promise|:
         </p>
         <ol class="algorithm">
+          <li>If |promise| was passed:
+            <ol>
+              <li>Let |global:Window| be |promise|'s [=relevant global object=].
+              </li>
+              <li>Let |document:Document| be |global|'s [=associated `Document`=].
+              </li>
+              <li>If |global| does not have [=transient activation=], [=queue a global task=] on
+              the [=user interaction task source=] given |global| to [=reject=] |promise| with a
+              "{{NotAllowedError}}" {{DOMException}} and abort these steps.
+              </li>
+              <li>Otherwise, [=consume user activation=] of |global|.
```

Comment:
I'm not sure this really works if "requesting permission to use" is called while in parallel as the transient activation check really has to happen in the main thread.